### DISCUSSION TITLE
Framework is constantly using the location

### DISCUSSION THREAD

**Discussion by Stophface** [rnmapbox/maps#2900] [Q&A]
I am developing an App which - among other things - tracks your location. The user is asked to give location permission **always**. Always means the App is allowed to access the location, even when the App is brought to the background. 

I am using `rnmapbox` to show the user location on a map. The `<Mapbox.MapView />` is implemented in a ` TabNavigator`  from [reactnavigation](https://reactnavigation.org/). This means, that once a component has been mounted inside the `TabNavigator` (like the `<Mapbox.MapView />`), it never gets unmounted, until the App is terminated. Currently I am testing the App on a real device and I am observing the following behaviour. Once the user has granted access to use the location `Always`  and the user has opened the `<Mapbox.MapView />` Tab and brings the App to the background (without terminating it), the App still uses the location. This also happens, when the user changes the Tab.  I assuem this is because `<Mapbox.MapView />` needs the location in order to show `<Mapbox.UserLocation />`  and once the user has granted the permission `Always`, `<Mapbox.MapView />` cconstantly uses the location. 

I expect `<Mapbox.UserLocation />` or `<Mapbox.MapView />` to not always use the location. I expect there to be some sort of a toggle or flag that I can set in order to turn off the accessing of the users locatin by the framework. 
Is there something like this, or is this a bug?

**Comment by mfazekas** [comment]
We don't listen to react-navigation onFocus and onBlur events. You can handle those events yourself and remove UserLocation when https://reactnavigation.org/docs/use-focus-effect the view goes out of focus. See also the `requestsAlwaysUse` on `UserLocation`

**Reply by Stophface** [reply]
I am using `useIsFocused()` from [react navigation](https://reactnavigation.org/docs/use-is-focused/) and conditionally render the `UserLocation`, but it is still using the location, even when the App is in the background or the tab is not focused. I am logging `isFocused` to validate if it is correct or not - and it is.

But am I assuming it correct that when setting `requestsAlwaysUse` to `false`  on the `UserLocation`: This would result in the Framework not using the location when the App is in the background, eventhough it has the permissions to do so?
